src/Microsoft.Health.Fhir.Shared.Api/Features/Resources/Bundle/BundleHandler.cs

```diff
@@ -97,6 +99,8 @@ private async Task FillRequestLists(List<Hl7.Fhir.Model.Bundle.EntryComponent> b
                 httpContext.Response.Body = new MemoryStream();
 
                 var requestUri = new Uri(_fhirRequestContextAccessor.FhirRequestContext.BaseUri, entry.Request.Url);
+                httpContext.Request.Scheme = requestUri.Scheme;
```

Comment:
would the entry.Request.Url always be relative URL (and if not do we have validation?)

Reply:
According to the spec the `entry.Request.Url` is "The URL for this entry, relative to the root (the address to which the request is posted)." See [bundle spec](https://www.hl7.org/fhir/bundle-definitions.html#Bundle.entry.request.url) for details.

Reply:
Also, how does it handle correlationId? We are not passing by the HTTP header. Is it because this is within the same transient session and IoC contains returns the same instance of CorrelationIdProvider?

Reply:
It's using JsonContenType below is that something we can assume? If the incoming request was Xml, would this still work?

Reply:
> Also, how does it handle correlationId? 

We set the correlationId on the FhirRequestContext that we pass to the sub calls.

Reply:
> It's using JsonContenType below is that something we can assume? If the incoming request was Xml, would this still work?

Once it's within the context of the main bundle call we're only working with the pocos. We can choose from there to send all the subsequent requests as json. There's a decent amount of parsing/serializing going on here that we could optimize by not relying on the http request.